contracts/l2/dai.sol

```diff
@@ -96,17 +93,19 @@ contract Dai {
 
   // --- ERC20 Mutations ---
   function transfer(address to, uint256 value) external returns (bool) {
+    require(to != address(0) && to != address(this), "Dai/invalid-address");
     uint256 balance = balanceOf[msg.sender];
     require(balance >= value, "Dai/insufficient-balance");
 
     balanceOf[msg.sender] = balance - value;
-    balanceOf[to] = add(balanceOf[to], value);
+    balanceOf[to] += value;
```

Comment:
just to double check, where is the overflow checked on this? (z >= x)

Reply:
I'll run a fuzzing campaign to check on this, probably improving the current echidna tests to check for overflows.